File: 0202-happy-number/0202-happy-number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        seen = set()

        while n != 1:
            total = 0
            x = n

            while x > 0:
                d = x % 10
                total += d * d
                x //= 10

            if total in seen:
                return False

            seen.add(total)
            n = total

        return True

File: 0202-happy-number/0202-happy-number.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isHappy = function(n) {
    const seen = new Set();
    while (n !== 1) 
    {
        let total = 0;
        let x = n;


        while (x > 0) 
        {
            let d = x % 10;
            total += d * d;
            x = Math.floor(x / 10);
        }

        if (seen.has(total)) return false;

        seen.add(total);
        n = total;
    }


    return true;
};